bridge/README.md

```diff
@@ -0,0 +1,96 @@
+
+Polygon PoS bridge provides a bridging mechanism that is near-instant, low-cost, and quite flexible.
+
+There is no change to the circulating supply of your token when it crosses the bridge:
+- Tokens that leave the Ethereum network are locked and the same number of tokens are minted on Polygon PoS as a pegged token (1:1).
+- To move the tokens back to the Ethereum network, tokens are burned on Polygon PoS network and unlocked on Ethereum network.
+
+## Listener
+
+The bridge module has a `BaseListner` which is extended by `RootChainListener`, `MaticChainListener` and `HeimdallListener`. It has all the methods required to start polling, listening to incoming headers(blocks) and stopping the process if required. All the 3 listneres uses these properties with their individual implementations on how to handle the incoming header once received.
+
+For example in `RootChainListner` the incoming header is used to determine the current height of root chain and calculate the `from` and `to` block numbers using which the events are fetched from the root chain. These events are then sent to `handleLog` where based on their event signature they are added to queue as tasks for further processing by their respective processors.
+
+## Processor
+
+There is a `BaseProcessor` which is extended by every processor in the processor package. It has all the methods required to start processing the events, registering for the tasks which are there in the queue and stopping the process if required. All the processors uses these properties with their individual implementations on how to handle the incoming events once received.
+
+Once the event is added to the queue by the `Listener`, the `Processor` takes over and processes the events which are added into the queue based on their event signature. Each processor has `RegisterTasks` method using which they register to process specific tasks added to the queue based on the module they serve. For example `StakingProcessor` registers for `Staking` related tasks, `ClerkProcessor` registers for `Clerk` related tasks and so on. You can look into each processor to check which tasks they are registered for.
+
+## How to start bridge
+
+Bridge should used by validator nodes only as they are the only ones who can send txns on heimdall chain, So if a non validator node is running bridge then it's of no use as it won't be able to send txns to heimdall chain.
+
+To start bridge as a validator you just have to add `--bridge` command to your heimdall node command, You can also control which services you want to run from processor by using `--all` or `--only` flags.
+
+To run all the service :
+
+```bash
+./heimdalld start --bridge --all
+```
+
+To use specific services :
+
+```bash
+./heimdalld start --bridge --only=clerk,staking
+```
+
+## Reset
+
+> :warning: Do this only when you are advised so and you understand the impact of this command. 
+
+If you want to reset the bridge server data, you can use the following command:
```

Comment:
It should be all the heimdall data, right?

Reply:
No, It cleans only bridge db